trie/node.go

```diff
@@ -53,9 +54,11 @@ var nilValueNode = valueNode(nil)
 
 // EncodeRLP encodes a full node into the consensus RLP format.
 func (n *fullNode) EncodeRLP(w io.Writer) error {
-	eb := rlp.NewEncoderBuffer(w)
-	n.encode(eb)
-	return eb.Flush()
+	h := newHasher(false)
+	defer returnHasherToPool(h)
+	h.tmp = n.encode(h.tmp[:0])
+	_, err := w.Write(h.tmp)
```

Comment:
To me, this seems like a very strange way to organize the code. Essentially, a `fullNode` is a very small thing, essentially a list of nodes. It's `EncodeRLP` implementation ought to be pretty straight-forward, just encoding the data. But this method accesses a `sync.Pool` to obtain a `hasher`. A hasher:

> hasher is a type used for the trie Hash operation. A hasher has some
> internal preallocated temp space

Which is odd, since, we're not actually hashing anything, we're just encoding. And what the code _does_, is that it just uses the hasher for the `preallocated temp space`. 

I think it's way too hacky for production use.

Reply:
Agree. This change should be left out.